monitoring/grafana/KubeStellarControllersMonitoring.json

```diff
@@ -0,0 +1,837 @@
+                "readOnly": true
+              }
+            },
+            "properties": [
+              {
+                "id": "custom.hideFrom",
+                "value": {
+                  "legend": false,
+                  "tooltip": false,
+                  "viz": true
+                }
+              }
+            ]
+          }
+        ]
+      },
+      "gridPos": {
+        "h": 8,
+        "w": 12,
+        "x": 0,
+        "y": 0
+      },
+      "id": 1,
+      "options": {
+        "legend": {
+          "calcs": [],
+          "displayMode": "list",
+          "placement": "bottom",
+          "showLegend": true
+        },
+        "tooltip": {
+          "mode": "single",
+          "sort": "none"
+        }
+      },
+      "targets": [
+        {
+          "datasource": {
+            "type": "prometheus",
+            "uid": "prometheus-test"
+          },
+          "disableTextWrap": false,
+          "editorMode": "code",
+          "exemplar": false,
+          "expr": "histogram_quantile(0.99, sum by(instance, name, le) (rate(workqueue_queue_duration_seconds_bucket{namespace=\"$SpaceNS\", job=\"$KSController\"}[5m])))",
+          "fullMetaSearch": false,
+          "includeNullMetadata": true,
+          "instant": false,
+          "interval": "",
+          "legendFormat": "__auto",
+          "range": true,
+          "refId": "A",
+          "useBackend": false
+        },
+        {
+          "datasource": {
+            "type": "prometheus",
+            "uid": "prometheus-test"
+          },
+          "editorMode": "code",
+          "expr": "histogram_quantile(0.50, sum by(instance, name, le) (rate(workqueue_queue_duration_seconds_bucket{namespace=\"$SpaceNS\", job=\"$KSController\"}[5m])))",
+          "hide": false,
+          "instant": false,
+          "legendFormat": "__auto",
+          "range": true,
+          "refId": "B"
+        },
+        {
+          "datasource": {
+            "type": "prometheus",
+            "uid": "prometheus-test"
+          },
+          "editorMode": "code",
+          "expr": "histogram_quantile(0.25, sum by(instance, name, le) (rate(workqueue_queue_duration_seconds_bucket{namespace=\"$SpaceNS\", job=\"$KSController\"}[5m])))",
+          "hide": false,
+          "instant": false,
+          "legendFormat": "__auto",
+          "range": true,
+          "refId": "C"
+        },
+        {
+          "datasource": {
+            "type": "prometheus",
+            "uid": "prometheus-test"
+          },
+          "editorMode": "code",
+          "expr": "histogram_quantile(0.75, sum by(instance, name, le) (rate(workqueue_queue_duration_seconds_bucket{namespace=\"$SpaceNS\", job=\"$KSController\"}[5m])))",
+          "hide": false,
+          "instant": false,
+          "legendFormat": "__auto",
+          "range": true,
+          "refId": "D"
+        }
+      ],
+      "title": "Percentile of Sync Duration for Items in the Workqueue",
```

Comment:
This histogram has a wildly suboptimal set of buckets. To be fixed in a separate PR.

workqueue_queue_duration_seconds_bucket{container="transport-controller", endpoint="metrics", instance="10.244.0.38:8090", job="ks-transport-controller", le="1e-08", name="transport-controller", namespace="wds1-system", pod="transport-controller-7d55cc46c9-mpht2", service="ks-transport-controller"}
	0
workqueue_queue_duration_seconds_bucket{container="transport-controller", endpoint="metrics", instance="10.244.0.38:8090", job="ks-transport-controller", le="1e-07", name="transport-controller", namespace="wds1-system", pod="transport-controller-7d55cc46c9-mpht2", service="ks-transport-controller"}
	0
workqueue_queue_duration_seconds_bucket{container="transport-controller", endpoint="metrics", instance="10.244.0.38:8090", job="ks-transport-controller", le="1e-06", name="transport-controller", namespace="wds1-system", pod="transport-controller-7d55cc46c9-mpht2", service="ks-transport-controller"}
	0
workqueue_queue_duration_seconds_bucket{container="transport-controller", endpoint="metrics", instance="10.244.0.38:8090", job="ks-transport-controller", le="9.999999999999999e-06", name="transport-controller", namespace="wds1-system", pod="transport-controller-7d55cc46c9-mpht2", service="ks-transport-controller"}
	0
workqueue_queue_duration_seconds_bucket{container="transport-controller", endpoint="metrics", instance="10.244.0.38:8090", job="ks-transport-controller", le="9.999999999999999e-05", name="transport-controller", namespace="wds1-system", pod="transport-controller-7d55cc46c9-mpht2", service="ks-transport-controller"}
	0
workqueue_queue_duration_seconds_bucket{container="transport-controller", endpoint="metrics", instance="10.244.0.38:8090", job="ks-transport-controller", le="0.001", name="transport-controller", namespace="wds1-system", pod="transport-controller-7d55cc46c9-mpht2", service="ks-transport-controller"}
	0
workqueue_queue_duration_seconds_bucket{container="transport-controller", endpoint="metrics", instance="10.244.0.38:8090", job="ks-transport-controller", le="0.01", name="transport-controller", namespace="wds1-system", pod="transport-controller-7d55cc46c9-mpht2", service="ks-transport-controller"}
	0
workqueue_queue_duration_seconds_bucket{container="transport-controller", endpoint="metrics", instance="10.244.0.38:8090", job="ks-transport-controller", le="0.1", name="transport-controller", namespace="wds1-system", pod="transport-controller-7d55cc46c9-mpht2", service="ks-transport-controller"}
	11
workqueue_queue_duration_seconds_bucket{container="transport-controller", endpoint="metrics", instance="10.244.0.38:8090", job="ks-transport-controller", le="1", name="transport-controller", namespace="wds1-system", pod="transport-controller-7d55cc46c9-mpht2", service="ks-transport-controller"}
	1751
workqueue_queue_duration_seconds_bucket{container="transport-controller", endpoint="metrics", instance="10.244.0.38:8090", job="ks-transport-controller", le="10", name="transport-controller", namespace="wds1-system", pod="transport-controller-7d55cc46c9-mpht2", service="ks-transport-controller"}
	2308
workqueue_queue_duration_seconds_bucket{container="transport-controller", endpoint="metrics", instance="10.244.0.38:8090", job="ks-transport-controller", le="+Inf", name="transport-controller", namespace="wds1-system", pod="transport-controller-7d55cc46c9-mpht2", service="ks-transport-controller"}

Reply:
Making note of this issue. I will fix it in a new PR. Thanks!

Reply:
That might not be easy.

In the meantime, I think that --- since the histogram bins are so coarse --- it would be helpful to add the average to the graph.